spring-ai-core/src/main/java/org/springframework/ai/chat/history/ChatHistory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.chat.history;
+
+import java.util.List;
+
+/**
+ * Core abstraction for storing and retrieving chat history.
+ *
+ * @author Christian Tzolov
+ */
+public interface ChatHistory {
+
+	void add(ChatExchange historyGroup);
+
+	List<ChatExchange> get(String sessionId);
```

Comment:
```suggestion
	List<ChatExchange> get(String sessionId, int lastN);
```

the limit will want to be a pushed-down predicate, e.g. in non-memory impls.